src/reflect/scala/reflect/internal/Trees.scala

```diff
@@ -1910,11 +1910,11 @@ trait Trees extends api.Trees {
   implicit val UnApplyTag             = ClassTag[UnApply](classOf[UnApply])
   implicit val ValDefTag              = ClassTag[ValDef](classOf[ValDef])
   implicit val ValOrDefDefTag         = ClassTag[ValOrDefDef](classOf[ValOrDefDef])
-
-  val treeNodeCount = Statistics.newView("#created tree nodes")(nodeCount)
 }
 
-object TreesStats {
-  // statistics
+trait TreesStats {
+  // Nodecount is only accessible from within `Trees.scala`, so this is mixed in at `SymbolTable`
+  var nodeCount: Int
```

Comment:
This could be an abstract getter and setter.

But how about moving the abstract `val symbolTable` into `ReflectStats` and using a `self: ReflectStats =>` here, then you could just use `symbolTable.nodeCount` on the next line, and get rid of the `override var nodeCount`.

Reply:
I fixed this in a similar way: I added an abstract val `symbolTable` in `Trees.scala` and use `symbolTable.nodeCount`. That allows me to keep `ReflectStats` defined in `SymbolTable`, avoid the cyclic dependency and only override that in the call-site. Thanks for the suggestion!